src/home/homework.c

```diff
@@ -836,9 +837,68 @@ static int user_record_compile_effective_passwords(
         return 0;
 }
 
+static int determine_default_storage(UserStorage *ret) {
+        UserStorage storage = _USER_STORAGE_INVALID;
+        const char *e;
+        int r;
+
+        assert(ret);
+
+        /* homed tells us via an environment variable which default storage to use */
+        e = getenv("SYSTEMD_HOME_DEFAULT_STORAGE");
+        if (e) {
+                storage = user_storage_from_string(e);
+                if (storage < 0)
+                        log_warning("$SYSTEMD_HOME_DEFAULT_STORAGE set to invalid storage type, ignoring: %s", e);
```

Comment:
Dunno, this would be an internal programming error. Shouldn't this result in hard failure?

Reply:
well, users might set the env var in the service file. the config file will override it then if it is set. but if it isn't set it will just be propagated down the process tree and homework will honour it. While I wouldn't advertise that this works I'd still cater enough for it to work sensibly...